sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -81,19 +82,20 @@ def __init__(
     ) -> None:
         self.client_connection = client_connection
         self.id = id
-        self._properties = properties
         self.database_link = database_link
         self.container_link = "{}/colls/{}".format(database_link, self.id)
         self._is_system_key: Optional[bool] = None
         self._scripts: Optional[ScriptsProxy] = None
+        if properties:
+            self.client_connection.container_properties_cache[self.container_link] = _set_properties_cache(properties)
 
     def __repr__(self) -> str:
         return "<ContainerProxy [{}]>".format(self.container_link)[:1024]
 
     async def _get_properties(self) -> Dict[str, Any]:
-        if self._properties is None:
-            self._properties = await self.read()
-        return self._properties
+        if self.container_link not in self.client_connection.container_properties_cache:
+            await self.read()
+        return self.client_connection.container_properties_cache[self.container_link]
```

Comment:
for _get_properties -> this should always call self.read()? because we should always return all collection properties? (like index policy)

Reply:
For properties that might change like index policy it might be better to get the properties from a read, with this cache we reuse the ones we wont have to change to avoid having to do read calls on them, in particular partition key definition would be one the ones called the most.

Reply:
what are the returns for _get_properties as of today? does it include all properties or just partitionKeyDefinition?

Reply:
As of today it essentially returns the container dictionary that was cached from the container instance. This includes everything you would have gotten from calling a read for the container, but won't have the latest information if some properties had changed, and only lasts for the lifetime of the container instance.